### PR TITLE
[WFLY-4697] Use suspend/resume to send out module unavailable/available messages to EJBClient

### PR DESCRIPTION
**Pull request by rachmatowicz**
This PR makes the following changes:
- uses suspend/resume to send module availability/unavailablity messages to connected EJBClients to inform them that the server is suspending/resuming
  - fixes a related issue with cache topology timeouts when invocations which are redirected to another node try to commit as the node on which the original invocation was made is shutting down
  - reinstates RemoteFailoverTestCase which was failing due to problems with EJBClients making invocations on nodes as they were shutting down

Please see https://issues.jboss.org/browse/WFLY-4697 for more details.

### CODE REVIEW

**Comment by rachmatowicz** [comment]
Feedback appreciated.

File: clustering/infinispan/extension/src/main/java/org/jboss/as/clustering/infinispan/subsystem/TransactionResourceDefinition.java [hunk 1]
```diff
@@ -91,7 +91,7 @@
             .setMeasurementUnit(MeasurementUnit.MILLISECONDS)
             .setAllowExpression(true)
             .setFlags(AttributeAccess.Flag.RESTART_ALL_SERVICES)
-            .setDefaultValue(new ModelNode().set(30000))
+            .setDefaultValue(new ModelNode().set(10000))
```
Comment (pferraro) [inline clustering/infinispan/extension/src/main/java/org/jboss/as/clustering/infinispan/subsystem/TransactionResourceDefinition.java]: This needs a transformer.  See https://github.com/wildfly/wildfly/blob/master/clustering/common/src/main/java/org/jboss/as/clustering/controller/transform/DefaultValueAttributeConverter.java

File: testsuite/integration/clustering/src/test/java/org/jboss/as/test/clustering/cluster/ejb/remote/RemoteFailoverTestCase.java [hunk 2]
```diff
@@ -294,7 +299,10 @@ public void testConcurrentFailover() throws Exception {
                 Future<?> future = executor.scheduleWithFixedDelay(new IncrementTask(bean, count, latch), 0, INVOCATION_WAIT, TimeUnit.MILLISECONDS);
                 latch.await();
 
-                undeploy(this.findDeployment(target));
+                String deployment = this.findDeployment(target);
+                System.out.println("--- Undeploying: " + deployment + " ---");
+                undeploy(deployment);
+                // undeploy(this.findDeployment(target));
```
Comment (rhusar) [inline testsuite/integration/clustering/src/test/java/org/jboss/as/test/clustering/cluster/ejb/remote/RemoteFailoverTestCase.java]: Are these intentional? If anything you could put the loggin in the org.jboss.as.test.clustering.cluster.ClusterAbstractTestCase#undeploy method.

File: clustering/infinispan/extension/src/main/java/org/jboss/as/clustering/infinispan/subsystem/TransactionResourceDefinition.java (hunk 1 above)
Comment (rachmatowicz) [inline clustering/infinispan/extension/src/main/java/org/jboss/as/clustering/infinispan/subsystem/TransactionResourceDefinition.java]: Done.

**Comment by rachmatowicz** [comment]
retest please

**Pull request merged**